### ISSUE TITLE
Add edge case tests for TextCNN model (empty SMILES, long sequences, invalid characters)

### ISSUE THREAD

**Issue by Sainava** [deepchem/deepchem#4634]
## Summary
The TextCNN model (`deepchem/models/torch_models/text_cnn.py`) currently lacks test coverage for edge cases that users may encounter when working with real-world molecular datasets. This proposal adds three critical test cases to improve robustness and reliability.

## Current Test Coverage
The existing tests in `deepchem/models/torch_models/tests/test_textcnn_model.py` cover:
- ✅ Basic model instantiation (`test_textcnn_base`)
- ✅ Forward pass with well-formed inputs (`test_textcnn_base_forward`)
- ✅ Overfitting on small datasets (`test_textcnn_overfit_classification/regression`)
- ✅ Model serialization/reload (`test_textcnn_reload`)
- ✅ TensorFlow implementation parity (`test_textcnn_compare_with_tf_impl`)

### 1. **Empty/Very Short SMILES Strings**
**Current Behavior:** Unknown - not tested
**Expected Behavior:** Should handle gracefully or raise informative error

**Example cases:**
- Empty string: `""`
- Single atom: `"C"` 
- Minimal molecule: `"CC"`

**Why this matters:** Users might have datasets with missing or malformed entries. The model should handle these predictably.

### 2. **Very Long SMILES Sequences**
**Current Behavior:** `build_char_dict()` uses `max_length * 1.2` (line 357), but behavior beyond this is untested
**Expected Behavior:** Should handle or truncate gracefully

**Example cases:**
- SMILES > 1000 characters (large polymers, dendrimers)
- Edge case at exactly `seq_length` boundary

**Why this matters:** Large molecules (proteins, polymers) can have very long SMILES representations. Need to validate padding/truncation behavior.

### 3. **Invalid Characters in SMILES**
**Current Behavior:** `smiles_to_seq()` raises `ValueError('character not found in dict')` (line 414)
**Code reference:**
```python
else:
    raise ValueError('character not found in dict')
```

**Problem: This error path is completely untested**

**Example cases:**

Unicode characters: "C€O"
Uncommon symbols: "C@#$%"
Whitespace variations
Why this matters: Real datasets can contain corrupted or non-standard SMILES. The error handling needs validation to ensure users get clear error messages.

**Proposed Solution**
Add three new test functions to (`test_textcnn_model.py`)
```python
@pytest.mark.torch
def test_textcnn_empty_and_short_smiles():
    """Test TextCNN behavior with empty and very short SMILES strings."""
    # Test empty, single char, and minimal molecules
    
@pytest.mark.torch
def test_textcnn_long_sequences():
    """Test TextCNN with extremely long SMILES sequences."""
    # Test sequences exceeding typical lengths
    
@pytest.mark.torch
def test_textcnn_invalid_characters():
    """Test TextCNN error handling for invalid SMILES characters."""
    # Verify ValueError is raised with appropriate message
```
## Contribution
I'm interested in implementing these tests and submitting a PR. Looking forward to feedback on this approach!

**Comment by AyusKumarPathak** [comment]
Hi DeepChem team 👋 @Sainava 

I’d like to work on adding edge case tests for the TextCNN model.
Plan:
1) Identify edge cases like empty SMILES, invalid inputs, extreme lengths.
2) Write pytest tests covering each of these cases.
3) Ensure tests fail before fix and succeed after.
4) Update docs to show expected outputs for edge cases.

I will raise a draft PR with test cases soon.

Please assign this issue if my approach is fine.

— AyusKumarPathak

**Comment by hharshhsaini** [comment]
Hi! I'd like to work on this issue.

I've reviewed the TextCNN implementation in `deepchem/models/torch_models/text_cnn.py` and the existing test coverage in `test_textcnn_model.py`. The proposed edge case tests make sense and would improve the model's robustness.

My plan:
1. Add `test_textcnn_empty_and_short_smiles()` - test empty strings, single atoms ("C"), and minimal molecules ("CC")
2. Add `test_textcnn_long_sequences()` - test SMILES exceeding `seq_length` and boundary conditions around `max_length * 1.2`
3. Add `test_textcnn_invalid_characters()` - verify `ValueError` is raised with clear messages for invalid characters

I'll ensure all tests follow the existing patterns with `@pytest.mark.torch` decorator and proper assertions.

Could you please assign this to me?

**Comment by hharshhsaini** [comment]
Reseolved the issue , and its ready to merge , please checckout !!
